pkg/common/token/store.go

```diff
@@ -61,18 +61,13 @@ func (b *BasicTokenStore) GetType() string {
 // GetToken implements TokenStore.
 func (b *BasicTokenStore) GetToken(ctx context.Context) (*Token, error) {
 	// This ensures lastToken not to be nil while b.lastTokenRefreshError == nil.
-	// Also, sync.Once guarantees there is only 1 execution across all goroutines.
-	// Thus, we can set the token without the lock.
+	// Also, sync.Once.Do guarantees there is only 1 execution across all goroutines, and it blocks others until ends.
+	// Thus, we can set the token without the lock and can assume either lastToken or lastTokenRefreshError should be set.
 	b.initialRefreshOnce.Do((func() {
 		b.refreshTokenWithoutLock(ctx)
 	}))
 	defer b.tokenLock.RUnlock()
 	b.tokenLock.RLock()
```

Comment:
maybe this is also not needed?

Reply:
Interesting, it's probably true.

Readers may see either an old token (someone already verified it's actually outdated) or an old error but this shouldn't be an issue?

Reply:
OK. then let's keep it.